src/tm2/tm-device.cpp

```diff
@@ -1440,12 +1460,21 @@ namespace librealsense
             }
             auto finish = duration<double, std::milli>(environment::get_instance().get_time_service()->get_time());
 
-            double device_ms = (double)response.llNanoseconds*1e-6;
-            auto device = duration<double, std::milli>(device_ms);
-            auto diff = duration<double, std::nano>(start + (finish - start)/2 - device);
-            device_to_host_ns = diff.count();
+            //If usb response takes too long, skip update. 0.25ms is 5% of 200Hz
+            if ((finish.count() - start.count()) / 2 < 0.25)
```

Comment:
I think it would be clearer if you used `std::chrono::milliseconds(0.25) instead of the bare constant.  It would then be independent of the durations actual units.

Reply:
if you make this:
```C++
   if ((finish.count() - start.count()) / 2 > 0.25)
       continue;
```
Then you don't have to indent/change the meat of the computation.